src/Illuminate/Routing/Concerns/HasCustomAttributes.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Illuminate\Routing\Concerns;
+
+use Illuminate\Support\Collection;
+use ReflectionAttribute;
+use ReflectionMethod;
+
+trait HasCustomAttributes
+{
+    public function callAction($method, $parameters)
```

Comment:
A comment for clarity highlighting that the method is expected to work in a controller  could be helpful.
```suggestion
    /**
     * Call a controller action with custom attribute handling.
     *
     * @param string $method
     * @param array $parameters
     * @return mixed
     */
    public function callAction($method, $parameters)
```